dali/test/python/test_pipeline_multichannel.py

```diff
@@ -0,0 +1,274 @@
+        self.inputs = ops.ExternalSource()
+        self.should_crop = should_crop
+        self.should_resize = should_resize
+        self.should_transpose = should_transpose
+        self.should_normalize = should_normalize
+        if self.should_resize:
+            self.resize = ops.Resize(device = self.device,
+                                     resize_y = 900,
+                                     resize_x = 300,
+                                     min_filter=types.DALIInterpType.INTERP_LINEAR)
+        if self.should_crop:
+            self.crop = ops.Crop(device = self.device,
+                                 crop = (220, 224),
+                                 crop_pos_x = 0.3,
+                                 crop_pos_y = 0.2,
+                                 image_type = types.RGB)
+        if self.should_transpose:
+            self.transpose = ops.Transpose(device = self.device,
+                                           perm = (1, 0, 2))
+        if self.should_normalize:
+            self.cmn = ops.CropMirrorNormalize(device = self.device,
+                                               std = 255.,
+                                               mean = 0.,
+                                               output_layout = types.NHWC,
+                                               output_dtype = types.FLOAT)
+
+    def define_graph(self):
+        self.data = self.inputs()
+        sequence = self.data.gpu() if self.device == 'gpu' else self.data
+        out = sequence
+        if self.should_resize:
+            out = self.resize(out)
+        if self.should_crop:
+            out = self.crop(out)
+        if self.should_transpose:
+            out = self.transpose(out)
+        if self.should_normalize:
+            out = self.cmn(out)
+        return out
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data, layout=self.layout)
+
+class MultichannelSynthPythonOpPipeline(Pipeline):
+    def __init__(self, function, batch_size, layout, iterator, num_threads=1, device_id=0):
+        super(MultichannelSynthPythonOpPipeline, self).__init__(batch_size,
+                                                                num_threads,
+                                                                device_id,
+                                                                exec_async=False,
+                                                                exec_pipelined=False)
+        self.layout = layout
+        self.iterator = iterator
+        self.inputs = ops.ExternalSource()
+        self.oper = ops.PythonFunction(function=function)
+
+    def define_graph(self):
+        self.data = self.inputs()
+        out = self.oper(self.data)
+        return out
+
+    def iter_setup(self):
+        data = self.iterator.next()
+        self.feed_input(self.data, data, layout=self.layout)
+
+def check_resize_multichannel_vs_numpy(device, batch_size, shape):
```

Comment:
Can we aggregate these:
- check_resize_multichannel_vs_numpy
- check_crop_multichannel_vs_numpy
- check_transpose_multichannel_vs_numpy
- test_normalize_multichannel_vs_numpy
Into one test adding just one more loop that will set True only for one argument:
```
def check_resize_multichannel_vs_numpy(device, batch_size, shape, resize, crop, transpose, normalize):
    eii1 = RandomDataIterator(batch_size, shape=shape)
    eii2 = RandomDataIterator(batch_size, shape=shape)
    compare_pipelines(MultichannelSynthPipeline(device, batch_size, types.NHWC, iter(eii1), should_resize=resize, should_crop=crop, should_transpose=transpose, should_normalize=normalize),
                      MultichannelSynthPythonOpPipeline(resize_func, batch_size, types.NHWC, iter(eii2)),
                      batch_size=batch_size, N_iterations=10,
                      eps = 0.2)

def test_multichannel_vs_numpy():
    for mode in [(True, False, False, False), (False, True, False, False), (False, False, True, False), (False, False, False, True)]:
      for device in {'cpu', 'gpu'}:
          for batch_size in {1, 3}:
              for shape in {(2048, 512, 3), (2048, 512, 8)}:
                  resize, crop, transpose, normalize = mode
                  # TODO(janton) Transpose not implemented for CPU
                  if transpose and device == 'cpu:
                    continue
                  yield check_resize_multichannel_vs_numpy, device, batch_size, shape,  resize, crop, transpose, normalize
```

Reply:
I can but I don't think it helps in any way, when you run the tests you'll see `False, False, True, False` and you have no clue what went wrong. When you see `test_transpose_multichannel_vs_numpy` failed you know exactly what failed. Anyway I'll give it a try and see if it looks better

Reply:
There is one thing that your code is not taking care of and that is the different function that you pass to PythonOpPipeline. You have `resize_func` hardcoded. To make it work you'd need to select the function from the inputs you got, or pass it as another argument.

I'll see what I can do, but I think it might end up in something unreadable

Reply:
Done